### ISSUE TITLE
Game does not start on Wayland

### ISSUE THREAD

**Issue by kaimast** [openclonk/openclonk#137]
Hi,
I'm using Ubuntu 20.04 on Wayland. Usually X11/GLX games just work fine through XWayland. The binary form the openclonk package however does not start. Also there seems to be no .desktop file, so the only way to run it is via the command line.

```
kai@kai-thinkpad ~> openclonk
Command 'openclonk' is available in '/usr/games/openclonk'
The command could not be located because '/usr/games' is not included in the PATH environment variable.
openclonk: command not found
kai@kai-thinkpad ~ [127]> /usr/games/openclonk
[16:10:31] OpenClonk
[16:10:31] Version: 8.1 unix linux-x86_64 (5c6fadfb266d - 2018-03-16 18:25:33 +0100)
[16:10:31] ExePath: "/usr/games/"
[16:10:31] SystemDataPath: "/usr/share/games/openclonk/"
[16:10:31] UserDataPath: "/home/kai/.clonk/openclonk/"
[16:10:31] Command line: /usr/games/openclonk
[16:10:31] Couldn't find matching GLX visual
[16:10:31] Game cleared.
[16:10:31] Engine shut down.
```

The same problem is present with the flatpak.  
```
kai@kai-thinkpad ~ [1]> flatpak run org.openclonk.OpenClonk
[16:14:34] OpenClonk
[16:14:34] Version: 8.1 unix linux-x86_64 (5c6fadfb266d - 2018-03-16 18:25:33 +0100)
[16:14:34] ExePath: "/app/bin/"
[16:14:34] SystemDataPath: "/app/share/games/openclonk/"
[16:14:34] UserDataPath: "/home/kai/.clonk/openclonk/"
[16:14:34] Command line: openclonk
[16:14:34] Couldn't find matching GLX visual
[16:14:34] Game cleared.
[16:14:34] Engine shut down.
```

**Comment by lluchs** [comment]
Yes, unfortunately, this broke when SDL2 started to use Wayland per default. The game should work with XWayland. You can either unset `WAYLAND_DISPLAY` or set `SDL_VIDEODRIVER=x11` to force running in XWayland. Of course, we should support Wayland eventually, but I personally don't run it on any of my systems yet.

@jcaesar This might be something we can fix in the flatpak by adjusting the environment variables?

>Also there seems to be no .desktop file, so the only way to run it is via the command line.

Not sure about this, the game does come with a desktop file. It might be a packaging problem, maybe open an issue with the ubuntu package maintainers?

**Comment by jcaesar** [comment]
@lluchs Ty for the poke, I set `SDL_VIDEODRIVER=x11` and the build just finished. https://flathub.org/builds/#/builders/32/builds/18739
@kaimast Could you do me a favour and check if an update for OpenClonk is available from flatpak? I'm unsure whether this is rolled out automatically. (Also, please report whether it's actually working, without messing with the environment.)

**Comment by kaimast** [comment]
@jcaesar thanks for the quick update. The desktop file seems to be visible now but I still can't seem to get the game to run, even when I set the environment variable manually.

**Comment by jcaesar** [comment]
I just noticed that I'd left a 60GB chunk of my HDD unprovisioned precisely for this kind of thing and promptly installed myself a ubuntu. A few observations:
 - Both the Ubuntu package and the flatpak package come with a desktop file, but one may have to add some path to XDG_DATA_DIR for the flatpak one to work. (They appear in the search menu. Not on the desktop.)
 - I can start openclonk just fine with the `openclonk` command. Maybe you need to be in the games group or so?
 - Setting `WAYLAND_DISPLAY=` and `SDL_VIDEODRIVER=x11` does nothing.
 - Setting `SDL_VIDEODRIVER=wayland` changes the way it fails: [OpenClonk.log](https://github.com/openclonk/openclonk/files/4552001/OpenClonk.log.txt) (Hence I'm unsure about the "SDL2 started to use Wayland per default")

Other than that, Xwayland is running, DISPLAY is set, glxgears works… @lluchs Any ideas what to try?

**Comment by jcaesar** [comment]
Place that fails: https://github.com/openclonk/openclonk/blob/8b9f6930363d34b99982d5ae43782cfa64079ee4/src/platform/C4WindowSDL.cpp#L84

**Comment by jcaesar** [comment]
So setting `MultiSampling=1` in the `[Graphics]` section of `~/.clonk/openclonk/config` makes it start, but the performance is pretty terrible.

This is also odd:
```
[15:44:38] GL 3.3 (Core Profile) Mesa 20.0.5 on llvmpipe (LLVM 9.0.1, 256 bits) (VMware, Inc.)
```

**Comment by jcaesar** [comment]
@kaimast The [workaround is easy](https://github.com/jcaesar/openclonk/commit/5e76b7c70e6acabbd1ced5878415832116cbade2). Before I add that, can I ask you to confirm that it works when turning `MultiSampling` off? (Or if you can, build OC with the patch enabled…)

**Comment by jcaesar** [comment]
Since I just merged #142, I'll close this. Feel free to report any feedback here nevertheless, @kaimast. And thanks for the initial report.
